pkgs/applications/graphics/gimp/default.nix

```diff
@@ -28,50 +32,73 @@
 , ghostscript
 , aalib
 , shared-mime-info
-, python2Packages
+, python3Packages
 , libexif
 , gettext
+, appstream-glib
+, libxslt
+, gobject-introspection
+, wrapGAppsHook
+, perl
 , xorg
 , glib-networking
 , libmypaint
 , gexiv2
 , harfbuzz
 , mypaint-brushes
+, dbus-glib
 , libwebp
 , libheif
+, gjs
 , libgudev
 , openexr
 , AppKit
 , Cocoa
-, gtk-mac-integration-gtk2
 }:
 
 let
-  inherit (python2Packages) pygtk wrapPython python;
+  python3 = python3Packages.python.withPackages (pp: with pp; [ pygobject3 ]);
 in stdenv.mkDerivation rec {
   pname = "gimp";
-  version = "2.10.14";
-
-  src = fetchurl {
-    url = "http://download.gimp.org/pub/gimp/v${lib.versions.majorMinor version}/${pname}-${version}.tar.bz2";
-    sha256 = "0m6wdnfvsxyhimdd4v3351g4r1fklllnbipbwcfym3h7q88hz6yz";
+  version = "unstable-2019-11-04";
+
+  # src = fetchurl {
+  #   url = "http://download.gimp.org/pub/gimp/v${lib.versions.majorMinor version}/${pname}-${version}.tar.bz2";
+  #   sha256 = "0wdcr8d2ink4swn5r4v13bsiya6s3xm4ya97sdbhs4l40y7bb03x";
+  # };
+
+  src = fetchFromGitLab {
+    domain = "gitlab.gnome.org";
+    owner = "GNOME";
+    repo = "gimp";
+    rev = "c033b3498f7a7115d78d7928a7d3bd34a4caf197";
+    sha256 = "0d6nylyzcxkf8yrvx2qxvn1gsp9qs68309sjy2gv8x7iid5fs8di";
   };
 
   nativeBuildInputs = [
     pkgconfig
+    meson
+    ninja
     intltool
     gettext
-    wrapPython
+    appstream-glib
+    libxslt
+    gobject-introspection
+    wrapGAppsHook
+    perl
   ];
 
   buildInputs = [
+    python3
```

Comment:
Think this won't work if it ends up in a shebang for darwin.

Reply:
Bah. Hoping for C wrappers before GIMP 3 is released.

Reply:
Good news! `makeBinaryWrapper` is a thing now: https://github.com/NixOS/nixpkgs/pull/124556